gnocchi/storage/__init__.py

```diff
@@ -443,6 +447,14 @@ def compute_and_store_timeseries(self, metric, measures):
             current_first_block_timestamp = None
         else:
             current_first_block_timestamp = ts.first_block_timestamp()
+            # log if backwindow is larger than ~4MB to anticipate potential
+            # performance issue
+            if self.enable_telemetry and ts.ts.size > 320000:
```

Comment:
Could you use `len(ts)`? It's a detail but it sounds cleaner.

Reply:
will do